applications/ContactStructuralMechanicsApplication/custom_strategies/custom_convergencecriterias/alm_frictional_mortar_criteria.h

```diff
@@ -205,38 +205,38 @@ class ALMFrictionalMortarConvergenceCriteria : public virtual  BaseMortarConverg
                     const double lambda_tangent = norm_2(tangent_lagrange_multiplier); 
                     
                     // The friction coefficient
-                    const double& mu = (it_node)->GetValue(WEIGHTED_FRICTION);
+                    const double& mu = it_node->GetValue(WEIGHTED_FRICTION);
                     
                     // Finally we compute the augmented tangent pressure
-                    const double& gt = (it_node)->FastGetSolutionStepValue(WEIGHTED_SLIP);
+                    const double& gt = it_node->FastGetSolutionStepValue(WEIGHTED_SLIP);
                     const double augmented_tangent_pressure = std::abs(scale_factor * lambda_tangent + tangent_factor * epsilon * gt) + mu * augmented_normal_pressure;
                     
-                    (it_node)->SetValue(AUGMENTED_TANGENT_CONTACT_PRESSURE, augmented_tangent_pressure); // NOTE: This value is purely for debugging interest (to see the "effective" pressure)
+                    it_node->SetValue(AUGMENTED_TANGENT_CONTACT_PRESSURE, augmented_tangent_pressure); // NOTE: This value is purely for debugging interest (to see the "effective" pressure)
                     
                     if (augmented_tangent_pressure <= 0.0) // TODO: Check if it is minor equal or just minor
                     {
-                        if ((it_node)->Is(SLIP) == true )
+                        if (it_node->Is(SLIP) == true )
                         {
-                            (it_node)->Set(SLIP, false);
+                            it_node->Set(SLIP, false);
                             #pragma omp atomic
```

Comment:
i am wondering if using openmp reduction for those variables would work here. I am not sure, but if it does it would probably be faster

Reply:
I think I already checked and didn't work